lib/rouge/lexers/ttcn3.rb

```diff
@@ -0,0 +1,165 @@
+    class TTCN3 < RegexLexer
+      title "TTCN3"
+      desc "The TTCN3 programming language (ttcn-3.org)"
+
+      tag 'ttcn3'
+      filenames '*.ttcn', '*.ttcn3'
+      mimetypes 'text/x-ttcn3', 'text/x-ttcn'
+
+      def self.keywords
+        @keywords ||= %w(
+          module import group type port component signature external
+          execute const template function altstep testcase var timer if
+          else select case for while do label goto start stop return
+          break int2char int2unichar int2bit int2enum int2hex int2oct
+          int2str int2float float2int char2int char2oct unichar2int
+          unichar2oct bit2int bit2hex bit2oct bit2str hex2int hex2bit
+          hex2oct hex2str oct2int oct2bit oct2hex oct2str oct2char
+          oct2unichar str2int str2hex str2oct str2float enum2int
+          any2unistr lengthof sizeof ispresent ischosen isvalue isbound
+          istemplatekind regexp substr replace encvalue decvalue
+          encvalue_unichar decvalue_unichar encvalue_o decvalue_o
+          get_stringencoding remove_bom rnd hostid send receive
+          setverdict
+        )
+      end
+
+      def self.reserved
+        @reserved ||= %w(
+          all alt apply assert at configuration conjunct const control
+          delta deterministic disjunct duration fail finished fuzzy from
+          history implies inconc inv lazy mod mode notinv now omit
+          onentry onexit par pass prev realtime seq setstate static
+          stepsize stream timestamp until values wait
+        )
+      end
+
+      def self.types
+        @types ||= %w(
+          anytype address boolean bitstring charstring hexstring octetstring
+          component enumerated float integer port record set of union universal
+        )
+      end
+
+      def self.builtins
+        @builtins ||= []
+      end
+
+      # optional comment or whitespace
+      ws = %r((?:\s|//.*?\n|/[*].*?[*]/)+)
+      id = /[a-zA-Z_]\w*/
+
+      state :inline_whitespace do
+        rule %r/[ \t\r]+/, Text
+        rule %r/\\\n/, Text # line continuation
+        rule %r(/(\\\n)?[*].*?[*](\\\n)?/)m, Comment::Multiline
+      end
+
+      state :whitespace do
+        rule %r/\n+/m, Text
+        rule %r(//(\\.|.)*?$), Comment::Single
+        mixin :inline_whitespace
+      end
+
+      state :expr_whitespace do
+        rule %r/\n+/m, Text
+        mixin :whitespace
+      end
+
+      state :statements do
+        mixin :whitespace
+        digit = /\d_+\d|\d/
+        bin_digit = /[01]_+[01]|[01]/
+        oct_digit = /[0-7]_+[0-7]|[0-7]/
+        hex_digit = /\h_+\h|\h/
+        rule %r/"/, Str, :string
+        rule %r/'(?:\\.|[^\\]|\\u[0-9a-f]{4})'/, Str::Char
+        rule %r/#{digit}+\.#{digit}+([eE]#{digit}+)?[fd]?/i, Num::Float
+        rule %r/'#{bin_digit}+'B/i, Num::Bin
+        rule %r/'#{hex_digit}+'H/i, Num::Hex
+        rule %r/'#{oct_digit}+'O/i, Num::Oct
+        rule %r/#{digit}+L?/i, Num::Integer
+        rule %r(\*/), Error
```

Comment:
Hello,

It's because in TTCN-3 an octet string format is such as 'CAFEDECA'O, an hexa string format is '0A0'H and an integer has the same format than in C (I based the TTCN-3 lexer on C lexer).

Yann

Reply:
I'm talking specifically about the rule on line 87. It matches an asterisk and produces an `Error` token.

Reply:
Got it.
I just updated the ttcn3.rb file